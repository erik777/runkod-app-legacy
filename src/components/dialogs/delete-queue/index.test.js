import React from 'react';
import {DialogContent} from './index';
import TestRenderer from 'react-test-renderer'; // ES6


test('1- default', () => {
  const props = {
    deleteQueue: {
      current: '',
      inProgress: false,
      files: [
        {
          parent: '/',
          label: 'file1.json'
        },
        {
          parent: '/',
          label: 'file2.json'
        }
      ],
      completed: [],
      failed: [],
      log: []
    },
    ui: {
      deleteDetail: false,
      deleteSummaryDetail: false
    },
    toggleUiProp: () => {
    },
    startDeleteQueue: () => {
    },
    resetDeleteQueue: () => {
    },
    fetchFiles: () => {
    }
  };

  const renderer = TestRenderer.create(
    <DialogContent {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});


test('2- show details', () => {
  const props = {
    deleteQueue: {
      current: '',
      inProgress: false,
      files: [
        {
          parent: '/',
          label: 'file1.json'
        },
        {
          parent: '/',
          label: 'file2.json'
        }
      ],
      completed: [],
      failed: [],
      log: []
    },
    ui: {
      deleteDetail: true,
      deleteSummaryDetail: false
    },
    toggleUiProp: () => {
    },
    startDeleteQueue: () => {
    },
    resetDeleteQueue: () => {
    },
    fetchFiles: () => {
    }
  };

  const renderer = TestRenderer.create(
    <DialogContent {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});


test('3- in progress', () => {
  const props = {
    deleteQueue: {
      current: '/file1.json',
      inProgress: true,
      files: [
        {
          parent: '/',
          label: 'file1.json'
        },
        {
          parent: '/',
          label: 'file2.json'
        }
      ],
      completed: [],
      failed: [],
      log: []
    },
    ui: {
      deleteDetail: false,
      deleteSummaryDetail: false
    },
    toggleUiProp: () => {
    },
    startDeleteQueue: () => {
    },
    resetDeleteQueue: () => {
    },
    fetchFiles: () => {
    }
  };

  const renderer = TestRenderer.create(
    <DialogContent {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});

test('4- completed', () => {
  const props = {
    deleteQueue: {
      current: '/file1.json',
      inProgress: false,
      files: [],
      completed: [{
        parent: '/',
        label: 'file1.json'
      }, {
        parent: '/',
        label: 'file2.json'
      }],
      failed: [],
      log: []
    },
    ui: {
      deleteDetail: false,
      deleteSummaryDetail: false
    },
    toggleUiProp: () => {
    },
    startDeleteQueue: () => {
    },
    resetDeleteQueue: () => {
    },
    fetchFiles: () => {
    }
  };

  const renderer = TestRenderer.create(
    <DialogContent {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});

test('5- completed. show detail', () => {
  const props = {
    deleteQueue: {
      current: '/file1.json',
      inProgress: false,
      files: [],
      completed: [{
        parent: '/',
        label: 'file1.json'
      }, {
        parent: '/',
        label: 'file2.json'
      }],
      failed: [],
      log: [{
        type: 'success',
        msg: '/file1.json deleted'
      }, {
        type: 'success',
        msg: '/file2.json deleted'
      }]
    },
    ui: {
      deleteDetail: false,
      deleteSummaryDetail: true
    },
    toggleUiProp: () => {
    },
    startDeleteQueue: () => {
    },
    resetDeleteQueue: () => {
    },
    fetchFiles: () => {
    }
  };

  const renderer = TestRenderer.create(
    <DialogContent {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});

test('6- show details with error', () => {
  const props = {
    deleteQueue: {
      current: '/file1.json',
      inProgress: false,
      files: [],
      completed: [{
        parent: '/',
        label: 'file1.json'
      }],
      failed: [ {
        parent: '/',
        label: 'file2.json'
      }],
      log: [{
        type: 'success',
        msg: '/file1.json deleted'
      }, {
        type: 'error',
        msg: '/file2.json could not delete'
      }]
    },
    ui: {
      deleteDetail: false,
      deleteSummaryDetail: true
    },
    toggleUiProp: () => {
    },
    startDeleteQueue: () => {
    },
    resetDeleteQueue: () => {
    },
    fetchFiles: () => {
    }
  };

  const renderer = TestRenderer.create(
    <DialogContent {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});