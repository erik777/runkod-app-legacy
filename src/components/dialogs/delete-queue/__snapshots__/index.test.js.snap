// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default 1`] = `
<div
  className="delete-confirm"
>
  <div
    className="title"
  >
    2 files will be deleted
  </div>
  <div
    className="details"
  >
    <span
      className="show-details"
      onClick={[Function]}
    >
      show details
    </span>
  </div>
  <div
    className="controls"
  >
    <button
      className="btn btn-primary btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
    <span
      className="separator"
    />
    <button
      className="btn btn-outline-info btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </div>
</div>
`;

exports[`2- show details 1`] = `
<div
  className="delete-confirm"
>
  <div
    className="title"
  >
    2 files will be deleted
  </div>
  <div
    className="details"
  >
    <span
      className="show-details"
      onClick={[Function]}
    >
      show details
    </span>
    <div
      className="detail-list"
    >
      <div
        className="detail-list-item"
      >
        <div
          className="index"
        >
          1
        </div>
        <div
          className="path"
        >
          /file1.json
        </div>
      </div>
      <div
        className="detail-list-item"
      >
        <div
          className="index"
        >
          2
        </div>
        <div
          className="path"
        >
          /file2.json
        </div>
      </div>
    </div>
  </div>
  <div
    className="controls"
  >
    <button
      className="btn btn-primary btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
    <span
      className="separator"
    />
    <button
      className="btn btn-outline-info btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </div>
</div>
`;

exports[`3- in progress 1`] = `
<div
  className="delete-progress"
>
  <div
    className="title"
  >
    Deleting
  </div>
  <div
    className="description"
  >
    /file1.json
  </div>
  <div
    className="controls"
  >
    <div
      className="progress"
    >
      <div
        aria-valuemax={2}
        aria-valuemin={0}
        aria-valuenow={0}
        className="progress-bar progress-bar-animated progress-bar-striped"
        role="progressbar"
        style={
          Object {
            "width": "0%",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`4- completed 1`] = `
<div
  className="delete-done"
>
  <div
    className="title"
  >
    Done
  </div>
  <div
    className="summary"
  >
    <span
      className="summary-item completed"
    >
      2 files deleted
    </span>
  </div>
  <div
    className="details"
  >
    <span
      className="show-details"
      onClick={[Function]}
    >
      show details
    </span>
  </div>
  <div
    className="controls"
  >
    <button
      className="btn btn-secondary btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
  </div>
</div>
`;

exports[`5- completed. show detail 1`] = `
<div
  className="delete-done"
>
  <div
    className="title"
  >
    Done
  </div>
  <div
    className="summary"
  >
    <span
      className="summary-item completed"
    >
      2 files deleted
    </span>
  </div>
  <div
    className="details"
  >
    <span
      className="show-details"
      onClick={[Function]}
    >
      show details
    </span>
    <div
      className="detail-list user-select"
    >
      <div
        className="detail-list-item success"
      >
        /file1.json deleted
      </div>
      <div
        className="detail-list-item success"
      >
        /file2.json deleted
      </div>
    </div>
  </div>
  <div
    className="controls"
  >
    <button
      className="btn btn-secondary btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
  </div>
</div>
`;

exports[`6- show details with error 1`] = `
<div
  className="delete-done"
>
  <div
    className="title"
  >
    Done
  </div>
  <div
    className="summary"
  >
    <span
      className="summary-item completed"
    >
      1 file deleted
    </span>
    <span
      className="summary-item failed"
    >
      1 file failed
    </span>
  </div>
  <div
    className="details"
  >
    <span
      className="show-details"
      onClick={[Function]}
    >
      show details
    </span>
    <div
      className="detail-list user-select"
    >
      <div
        className="detail-list-item success"
      >
        /file1.json deleted
      </div>
      <div
        className="detail-list-item error"
      >
        /file2.json could not delete
      </div>
    </div>
  </div>
  <div
    className="controls"
  >
    <button
      className="btn btn-secondary btn-sm"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
  </div>
</div>
`;
